src/stash/anchor.rs

```diff
@@ -331,6 +331,20 @@ impl Anchor {
     pub fn anchor_id(&self) -> AnchorId {
         self.clone().consensus_commit()
     }
+
+    pub fn conceal_except(&mut self, contract_id: ContractId) -> usize {
+        self.commitment.entropy = None;
+        self.commitment.commitments.iter_mut().fold(
+            0usize,
+            |mut count, item| {
+                if item.protocol != Some((*contract_id.as_slice()).into()) {
+                    item.protocol = None;
+                    count += 1;
+                }
+                count
```

Comment:
We are nullifying entropy and all other structures by concealing, except for provided contract id.

My reasoning: The anchor data for un-owned states will be never required the spender. So they are safe to be removed.

Is that correct?

Reply:
This prevents the new owner from knowing how many assets were transferred together with that particular state transition. He will not be able to recover this information and only previous owner will be able to disclose/know all other details of transfer.

It actually removes link/correlation analysis between different asset transfers